angular-semantic-ui.js

```diff
@@ -0,0 +1,2818 @@
+
+        scope.$watch( attributes[ attribute ], function(input)
+        {
+          // If the attribute value is a string, take it as the selector
+          if ( angular.isString( input ) ) 
+          {
+            settings.$ = input;
+          }
+          // If the attribute value is an object, overwrite the defaults.
+          else if ( angular.isObject( input ) ) 
+          {
+            if ( !angular.isString( input.evt ) ) input.evt = settings.evt;
+            if ( !angular.isDefined( input.enabled ) ) input.enabled = settings.enabled;
+
+            settings = input;
+          }
+
+          if ( previousEvent )
+          {
+            element.off( previousEvent, onEvent );
+          }
+
+          element.on( previousEvent = settings.evt, onEvent );
+
+        }, true );
+      },
+      watcher: function(scope, expression, func, context, force) 
+      {
+        var ignoreUpdate = false;
+
+        scope.$watch( expression, function( updated )
+        {
+          if ( !ignoreUpdate )
+          {
+            func.call( context, updated );
+          }
+
+          ignoreUpdate = false;
+        });
+
+        return {
+          set: function(value)
+          {
+            if ( scope[ expression ] != value || force )
+            {
+              scope.$evalAsync(function()
+              {
+                scope[ expression ] = value;
+                ignoreUpdate = true;
+              });
+            }
+          },
+          update: function()
+          {
+            scope.$evalAsync(function()
+            {
+              ignoreUpdate = true;
+            });
+          }
+        }
+      },
+      RecursiveCompiler: function(postLink)
+      {
+        return function(element, link)
+        {
+          // Normalize the link parameter
+          if( angular.isFunction( link ) )
+          {
+              link = { post: link };
+          }
+
+          // Break the recursion loop by removing the contents
+          var contents = element.contents().remove();
+          var compiledContents;
+
+          return {
+              pre: (link && link.pre) ? link.pre : null,
+              /**
+               * Compiles and re-adds the contents
+               */
+              post: function(scope, element)
+              {
+                  // Compile the contents
+                  if( !compiledContents ) 
+                  {
+                      compiledContents = $compile(contents);
+                  }
+
+                  // Re-add the compiled contents to the element
+                  compiledContents( scope, function(clone)
+                  {
+                      element.append(clone);
+                  });
+
+                  // Call the post-linking function, if any
+                  if ( link && link.post )
+                  {
+                      link.post.apply( null, arguments );
+                  }
+
+                  if ( angular.isFunction( postLink ) )
```

Comment:
If you have a better way to handle directives that contain themselves please let me know! I'm not as experienced as you obviously when it comes to Angular